### DISCUSSION TITLE
Horizon 7.8 Treadmill - Incline Adjustments Only Lowering Incline (Log Attached)

### DISCUSSION THREAD

**Discussion by bgtees** [cagnulein/qdomyos-zwift#2846] [General]
Have QZ installed on my iPhone (15PM, if it matters). Connects to my Horizon 7.8 treadmill just fine and it displays all of the metrics correctly on the dashboard. If I adjust the speed up or down from the QZ app, it works as expected. But, if I try to adjust the incline from the app, it only decreases the incline (as is hopefully shown in the attached debug log). If I have the incline manually set at 6.0 and press the "+" sign in the app, it will lower the incline to 3.0, and if I press the "-" sign in the app, that will also lower the incline even more. Any help with how to fix this would be greatly appreciated. Thanks!

[debug-Fri_Nov_29_23_01_06_2024.log](https://github.com/user-attachments/files/17964347/debug-Fri_Nov_29_23_01_06_2024.log)

**Comment by cagnulein** [comment]
Thanks I will check it ASAP

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno sab 30 nov 2024 alle 06:33 bgtees ***@***.***> ha
scritto:

**Comment by cagnulein** [comment]
got the issue, it's about this one https://github.com/cagnulein/qdomyos-zwift/issues/2495

the @lizzieac treadmill wants a /2 inclination to set the right one, that's why for your model i'm sending it in this way. @lizzieac are you still using QZ? maybe it will be useful to compare both firmware revision

let me know

**Comment by LizzieAC** [comment]
I am, however, I’ve been away and it’s been beautiful out so I’ve been going outside. I will try it out when I get home and send you back a report.Thank you for reaching out!Sent from my iPadOn Nov 30, 2024, at 8:09 AM, Roberto Viola ***@***.***> wrote:﻿
got the issue, it's about this one #2495
the @LizzieAC treadmill wants a /2 inclination to set the right one, that's why for your model i'm sending it in this way. @LizzieAC are you still using QZ? maybe it will be useful to compare both firmware revision
let me know

**Reply by cagnulein** [reply]
thanks @LizzieAC !
@bgtees could you please share your firmware version?

**Reply by bgtees** [reply]
> thanks @LizzieAC ! @bgtees could you please share your firmware version?

It's 5.004. Per the Horizon website, it's the current firmware version.

**Reply by cagnulein** [reply]
Ok maybe @LizzieAC has an outdated one? Let's wait for a confirmation so in case I will remove the workaround that I did for its firmware

**Reply by bgtees** [reply]
> Ok maybe @LizzieAC has an outdated one? Let's wait for a confirmation so in case I will remove the workaround that I did for its firmware

So until that change is made, can I just manually edit the numbers in the "Inclination Overrides" setting to get it working?

**Comment by cagnulein** [comment]
Yes it should work. I would like to do a changes that work for everyone

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 4 dic 2024 alle 19:25 bgtees ***@***.***> ha
scritto:

**Comment by LizzieAC** [comment]
I apologize for not testing.  I have been traveling quite a bit.  I’ll be back him after the 14th and I will get on and test. Sent from my iPadOn Dec 4, 2024, at 1:43 PM, Roberto Viola ***@***.***> wrote:﻿
Yes it should work. I would like to do a changes that work for everyone



Roberto Viola

Software engineer and open source enthusiast

http://robertoviola.cloud







Il giorno mer 4 dic 2024 alle 19:25 bgtees ***@***.***> ha

scritto:
